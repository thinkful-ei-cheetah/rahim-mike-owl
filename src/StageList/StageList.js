import React from 'react';
import Stage from '../Stage/Stage';

export default function StageList(props) {
  const list = props.store.participants;

  const onStageTrue = list.filter(element => element.onStage === true);
  const onStageList = onStageTrue.map(element => (
    <Stage name={element.name} avatar={element.avatar} key={element.id} />
  ));
  return (
    <div className='views-container views-container__with-sidebar'>
      {onStageList}
    </div>
  );
}
