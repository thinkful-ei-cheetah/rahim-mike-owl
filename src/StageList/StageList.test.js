import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import StageList from './StageList';
import store from '../STORE';

describe('StageList component', () => {
  it('should render without crashiing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<StageList store={store} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders UI with data', () => {
    const tree = renderer.create(<StageList store={store} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
