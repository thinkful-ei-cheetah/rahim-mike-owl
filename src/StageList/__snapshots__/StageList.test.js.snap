// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StageList component Renders UI with data 1`] = `
<div
  className="views-container views-container__with-sidebar"
>
  <div
    className="views-container views-container__with-sidebar"
  >
    <div>
      <p />
    </div>
    <div>
      <img />
    </div>
  </div>
  <div
    className="views-container views-container__with-sidebar"
  >
    <div>
      <p />
    </div>
    <div>
      <img />
    </div>
  </div>
</div>
`;
